codecs/avif_dec/build.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set -e
+
+export OPTIMIZE="-Os"
+export LDFLAGS="${OPTIMIZE}"
+export CFLAGS="${OPTIMIZE}"
+export CPPFLAGS="${OPTIMIZE}"
+
+echo "============================================="
+echo "Compiling libaom"
+echo "============================================="
+test -n "$SKIP_LIBAOM" || (
+  cd node_modules/libavif/ext  
+  test -d aom || git clone -b v1.0.0-errata1-avif --depth 1 https://aomedia.googlesource.com/aom aom
+
+  cd aom
+  mkdir -p build.libavif
+  cd build.libavif
+
+  emcmake cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DENABLE_CCACHE=0 \
+        -DAOM_TARGET_CPU=generic \
+        -DENABLE_DOCS=0 \
+        -DENABLE_TESTS=0 \
+        -DCONFIG_ACCOUNTING=1 \
+        -DCONFIG_INSPECTION=0 \
+        -DCONFIG_MULTITHREAD=0 \
+        -DCONFIG_RUNTIME_CPU_DETECT=0 \
+        -DCONFIG_WEBM_IO=0 \
+        ../
+
+  emmake make
+)
+echo "============================================="
+echo "Compiling libaom done"
+echo "============================================="
+
+echo "============================================="
+echo "Compiling libavif"
+echo "============================================="
+test -n "$SKIP_LIBAVIF" || (
+  cd node_modules/libavif
+  mkdir -p build
+  cd build
+  emcmake cmake \
+    DCMAKE_BUILD_TYPE=Release \
+    -DAVIF_CODEC_AOM=1 \
```

Comment:
Out of curiosity - why this codec and not some other(s)? (I don't know anything about AVIF, so this might be a stupid question.)

Reply:
No deeper reasoning. I just wanted to get _one_ to work to get started (and I do not know if they provide different capabilities).

Seems like since I started more impls have been added, as well. Feel free to experiment.